eng/build.sh

```diff
@@ -136,6 +136,21 @@ showSubsetHelp()
   "$scriptroot/common/build.sh" "-restore" "-build" "/p:Subset=help" "/clp:nosummary"
 }
 
+assertInstalledDependency()
+{
+  dependency="$(echo "$1" | awk '{print tolower($0)}')"
+  location="$(command -v $dependency)"
+
+  if [ -z "$location" ]; then
+    echo "$dependency is required to build this repo. Make sure to install it and try again."
+    echo "For a full list of requirements, see https://github.com/dotnet/runtime/blob/master/docs/workflow/requirements/linux-requirements.md"
+    exit 1
+  fi
+}
+
+assertInstalledDependency 'CMake'
+assertInstalledDependency 'Git'
```

Comment:
Git isn't a dependency. Sourcebuild needs to work without got.

Reply:
Actually, it is. The build uses it to extract it the commit hash that we embed into the binaries.

Reply:
Is it done unconditionally, or can this behavior be suppressed by configuration? I believe that there are configuration options to suppress this behavior.

Reply:
You are right, I've tried to uninstall git and the build completed ok, just no version info got embedded into the binaries.

Reply:
So we just check for CMake then?

Reply:
yes